pandas/core/tools/datetimes.py

```diff
@@ -343,8 +410,19 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                 # fallback
                 if result is None:
                     try:
-                        result = array_strptime(arg, format, exact=exact,
-                                                errors=errors)
+                        parsing_tzname = '%Z' in format
+                        parsing_tzoffset = '%z' in format
+                        if tz is not None and (parsing_tzname or
+                                               parsing_tzoffset):
+                            raise ValueError("Cannot pass a tz argument when "
+                                             "parsing strings with timezone "
+                                             "information.")
+                        result, tznames, tzoffsets = array_strptime(
+                            arg, format, exact=exact, errors=errors)
+                        if parsing_tzname or parsing_tzoffset:
```

Comment:
why do you need the parsing_tzoffset & parsing_tzname, won't tznames and tzoffsets be not None in this case? (and None if these are not found)?